### PR TITLE
Portfolio: preserve data in the browser

### PR DESCRIPTION
**Pull request by getssh**
We worked on adding local-storage 

- Added windows.local storage function to store data in the browser
- Used setItem and getItem methods to set and get data from the local storage
- Used JSON.stringify and JSON.parse for better interaction with local storage
- Files checked for linter errors and passed

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -197,4 +197,33 @@ function popClose() {
...
+    errorMsg.innerHTML = `Email address should be in lower case <b>${formEmail.value.toLowerCase()}</b>`;
+  }
+  const dataObj = {
+    name: fullName.value,
+    email: formEmail.value,
+    message: Message.value,
+  };
+
+  window.localStorage.setItem('userData', JSON.stringify(dataObj));
+});
```
Comment (brainconnect93) [inline index.js]: - [ ] While you make a great effort working on the preservation of data in the localStorage, I noticed when a user fills the form with their data it does not populate into the localStorage. I would kindly suggest you fix this issue and be sure the data fill by the user is populated to the localStorage.

  >N.B: 1. The contact form information is persistent to page reloads.
  >2. After filling out the form, and not submitting it, the entered values should be saved in the browser.
  >3. When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
  >4. If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.

Below is the picture of my claim 👇 

gess local

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :recycle:

### Hi @Team, 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 
-  Has a proper Pull Request. ✔️  
- No Linter errors :heavy_check_mark: 
- Good use of Github flow ✔️ 

### Changes Required

- [x] I would kindly advise you to work on the MIT license link in your README as this is part of the requirement that you should have a professional README file and all the links and necessary headers be written professionally. Currently, the MIT license link is broken kindly fix it.

MIT 👇 

mit

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js [hunk 2]
```diff
@@ -197,4 +197,36 @@ function popClose() {
...
+formEmail.value = userDataFromLocal.email;
+Message.value = userDataFromLocal.message;
+
+form.addEventListener('change', () => {
+  const dataObj = {
+    name: fullName.value,
+    email: formEmail.value,
+    message: Message.value,
+  };
+
```
Comment (Shedrack-Sunday) [inline index.js]: - [ ] While you did make efforts to fix this, the data is still not saved on local storage, and on reload, it disappears. 
- Instructions 
image
 - Please checkout this awesome [Resource ](https://dev.to/michaelburrows/how-to-save-data-in-localstorage-using-javascript-994) that has a step-by-step guide on how to fix this. 
 - Use your inspect on chrome developer tools to confirm just as I did in the view below :
 
 # Storage view :
 
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @getssh     👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯 



Highlights
✅All linters are checked
✅ Descriptive commit messages 
  
Required Changes ♻️

- [x] You did make effort in the  Project and you really did a great job 👏 . However, your form input seems to not be saved on localstorage on entry. Check the comments for the review proper. 

_check the comment section for a detailed review_

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: index.js (hunk 2 above)
Comment (getssh) [reply index.js]: Hello @Shedrack-Sunday, Thank you for the feedback.
After I changed the order of getItem and setItem and changed onchange event listener to keyup event listener, It's working and saving to localstorage as I type to the input fields 
Screenshot 2023-02-04 091246

Thank you

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 2 above)
Comment (Shedrack-Sunday) [reply index.js]: You're welcome

**Pull request merged**